matcher/service/openreview_interface.py

```diff
@@ -54,59 +54,34 @@ def _edge_to_score(edge, translate_map=None):
     return a numeric score, given an Edge.
     '''
 
-    score = edge.weight
+    score = edge['weight']
 
     if translate_map:
         try:
-            score = translate_map[edge.label]
+            score = translate_map[edge['label']]
         except KeyError:
             raise EncoderError(
                 'Cannot translate label {} to score. Valid labels are: {}'.format(
-                    edge.label, translate_map.keys()))
+                    edge['label'], translate_map.keys()))
 
     if not isinstance(score, float) and not isinstance(score, int):
         try:
             score = float(score)
         except ValueError:
             raise EncoderError(
-                'Edge {} has weight that is neither float nor int: {}, type {}'.format(
-                    edge.id, edge.weight, type(edge.weight)))
+                'Edge has weight that is neither float nor int: {}, type {}'.format(
+                    edge['weight'], type(edge['weight'])))
 
     return score
 
-def get_all_edges(client, edge_invitation_id, logger=None):
-    '''Helper function for retrieving and parsing all edges in bulk'''
-
-    all_edges = []
-    logger.debug('GET invitation id={}'.format(edge_invitation_id))
-    edge_invitation = client.get_invitation(edge_invitation_id)
-
-    edges_grouped_by_paper = client.get_grouped_edges(
-        invitation=edge_invitation_id,
-        groupby='head',
-        select='tail,label,weight'
-    )
-
-    logger.debug('GET grouped edges invitation id={}'.format(edge_invitation_id))
-    for group in edges_grouped_by_paper:
-        forum_id = group['id']['head']
-        for group_value in group['values']:
-            all_edges.append(build_edge(
-                edge_invitation,
-                forum_id,
-                group_value['tail'],
-                group_value.get('weight'),
-                group_value.get('label'),
-                None
-            ))
-    return all_edges
-
 class ConfigNoteInterface:
     def __init__(self, client, config_note_id, logger=logging.getLogger(__name__), cache=None):
         self.client = client
         self.config_note_id = config_note_id
         self.logger = logger
         self._cache = {} if not cache else cache
+        self._reviewers = None
```

Comment:
I'm sorry if I wasn't clear. `cache` doesn't make any sense. It has to be removed and all the values should be a  property of the class only when computed when  it is requested.  

Take a look to this: https://en.wikipedia.org/wiki/Lazy_initialization

Reply:
I liked the idea of lazy initialization. 
I am just starting to make these changes. Reviewers and Papers are the first vars to get lazy initializers and once every thing is working with the _cache I would get rid of that.

Let me know if you see any issues with the way I am implementing the lazy initializers.

Reply:
The class property _cache has now been removed.